internal/server/middlewares.go

```diff
@@ -84,3 +93,56 @@ func (e *EverestServer) checkOperatorUpgradeState(next echo.HandlerFunc) echo.Ha
 		return next(c)
 	}
 }
+
+func securityHeaders(oidcProvider *oidc.ProviderConfig) echo.MiddlewareFunc {
+	connectSrc := []string{CSPSelf}
+	if oidcProvider != nil {
+		connectSrc = append(connectSrc, oidcProvider.Issuer+oidc.WellKnownPath)
+		connectSrc = append(connectSrc, oidcProvider.TokenURL)
+	}
+
+	cspBuilder := cspbuilder.Builder{
+		Directives: map[string][]string{
+			cspbuilder.DefaultSrc: {CSPSelf},
+			cspbuilder.FontSrc:    {CSPSelf, "data:"},
+			cspbuilder.StyleSrc: {
+				CSPSelf,
+				// $NONCE will be replaced by the real nonce value
+				// auto-generated by the secure middleware. This nonce will
+				// be stored in the request's context and shall be embedded in
+				// the index.html template.
+				"$NONCE",
+				// @emotion adds an extra inline style with the SHA256 hash of
+				// an empty string, so we need to explicity allow it, see:
+				// https://github.com/emotion-js/emotion/issues/2996
+				"'sha256-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU='",
+			},
+			cspbuilder.FormAction:              {CSPSelf},
+			cspbuilder.BaseURI:                 {CSPSelf},
+			cspbuilder.ObjectSrc:               {CSPNone},
+			cspbuilder.FrameAncestors:          {CSPNone},
+			cspbuilder.UpgradeInsecureRequests: {},
+			cspbuilder.ConnectSrc:              connectSrc,
+		},
+	}
+
+	secureMiddleware := secure.New(secure.Options{
+		ContentSecurityPolicy:     cspBuilder.MustBuild(),
+		ContentTypeNosniff:        true,
+		CrossOriginEmbedderPolicy: "require-corp",
+		CrossOriginOpenerPolicy:   "same-origin",
+		CrossOriginResourcePolicy: "same-origin",
+		FrameDeny:                 true,
+		PermissionsPolicy:         PermissionsPolicy,
+		ReferrerPolicy:            "no-referrer",
+		// TODO (EVEREST-1180): Once we have native support for TLS, we should
+		// probably redirect all HTTP traffic to HTTPS.
+		// SSLRedirect:                   true,
+		// And we should set the HSTS header.
+		// STSIncludeSubdomains:          true,
+		// STSSeconds:                    31536000,
+		XPermittedCrossDomainPolicies: "none",
+	})
```

Comment:
In [the list](https://owasp.org/www-project-secure-headers/ci/headers_add.json) of the recommended headers there are also those two:
```
   {
      "name": "Cache-Control",
      "value": "no-store, max-age=0"
    },
    {
      "name": "Clear-Site-Data",
      "value": "\"cache\",\"cookies\",\"storage\""
    }
```
Are they not relevant in our case?

Reply:
To be honest I'm not 100% sure. I thought that users could benefit from the performance benefits of storing some content in a proxy server. Everest being a self-hosted solution maybe this isn't very relevant and we could just disable it.
@fabio-silva what do you think?

Reply:
After a deeper discussion with Fábio, we decided to [add them](https://github.com/percona/everest/pull/1195/commits/62b895c5462f3ebdc2937d0a51f30f377ddbe897). However, the `Clear-Site-Data` will only have `cache` and `cookies` and it won't have the `storage` because that would screw up our local storage where we store stuff like the token.